src/controllers/TransfersController.ts

```diff
@@ -79,4 +89,115 @@ export const TransfersController = {
       void reply.status(500)
     }
   },
+
+  transferByResource: async function (
+    request: FastifyRequest<{ Params: ITransferByResource; Querystring: ITransfer }>,
+    reply: FastifyReply,
+  ): Promise<void> {
+    const {
+      params: { resourceID },
+    } = request
+    const {
+      query: { page, limit, status },
+    } = request
+
+    try {
+      const transfers = await transfersService.findTransferByResourceID({ page, limit, status, resourceID: resourceID })
+
+      void reply.status(200).send(transfers)
+    } catch (e) {
+      if (e instanceof NotFound) {
+        void reply.status(404)
+      } else {
+        logger.error(e)
+        void reply.status(500)
+      }
+    }
+  },
+
+  transferBySourceDomainToDestinationDomain: async function (
+    request: FastifyRequest<{ Params: ITransferBySourceDomainToDestinationDomain; Querystring: ITransfer }>,
+    reply: FastifyReply,
+  ): Promise<void> {
+    const {
+      params: { sourceDomainID, destinationDomainID },
+    } = request
+    const {
+      query: { page, limit },
+    } = request
+
+    try {
+      const transfers = await transfersService.findTransferBySourceDomainToDestinationDomain({
+        page,
+        limit,
+        sourceDomainID: sourceDomainID,
+        destinationDomainID: destinationDomainID,
+      })
+
+      void reply.status(200).send(transfers)
+    } catch (e) {
+      if (e instanceof NotFound) {
+        void reply.status(404)
+      } else {
+        logger.error(e)
+        void reply.status(500)
+      }
+    }
+  },
+
+  transferByResourceBetweenDomains: async function (
+    request: FastifyRequest<{ Params: ITransferByResourceBetweenDomains; Querystring: ITransfer }>,
+    reply: FastifyReply,
+  ): Promise<void> {
+    const {
+      params: { resourceID, sourceDomainID, destinationDomainID },
+    } = request
+    const {
+      query: { page, limit },
+    } = request
+
+    try {
+      const transfers = await transfersService.findTransferByResourceBetweenDomains({
+        page,
+        limit,
+        resourceID: resourceID,
+        sourceDomainID: sourceDomainID,
+        destinationDomainID: destinationDomainID,
+      })
+
+      void reply.status(200).send(transfers)
+    } catch (e) {
+      if (e instanceof NotFound) {
+        void reply.status(404)
+      } else {
+        logger.error(e)
+        void reply.status(500)
+      }
+    }
+  },
+
+  transferByDomain: async function (
+    request: FastifyRequest<{ Params: ITransferByDomain; Querystring: ITransferByDomainQuery }>,
+    reply: FastifyReply,
+  ): Promise<void> {
+    const {
+      params: { domainID },
+    } = request
+    const {
+      query: { page, limit, status, domain },
```

Comment:
Validate the domain query param here

Reply:
There is a Fastify schema which you define so it automatically does this for you.
https://fastify.dev/docs/latest/Reference/Validation-and-Serialization/